libraries/aws_efs_file_system.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsEfsFileSystem < AwsResourceBase
+  name 'aws_efs_file_system'
+  desc 'Verifies settings for an EFS File System'
+
+  example "
+    describe aws_efs_file_system('fs-12345678') do
+      it                         { should be_encrypted }
+      its('size_in_bytes.value') { should cmp 6144 }
+    end
+
+    describe aws_efs_file_system(creation_token: 'my-token') do
+      its('encrypted')       { should cmp true }
+      its('throughput_mode') { should eq 'bursting' }
+    end
+  "
+
+  def initialize(opts = {})
+    # Assume the argument is a file_system_id when it is a string type object and reject any other type.
+    opts = { file_system_id: opts } if opts.is_a?(String)
+    opts = {} unless opts.is_a?(Hash)
+
+    # File system identifier can be either the file_system_id or the creation_token.
+    if opts[:file_system_id] && !opts[:file_system_id].empty?
+      raise ArgumentError, "#{@__resource_name__}: file_system_id must be in the format 'fs-' followed by 8 or 17 hexadecimal characters." unless /^fs\-([0-9a-f]{8})|(^fs\-[0-9a-f]{17})$/.match?(opts[:file_system_id])
+      file_system_arguments = { file_system_id: opts[:file_system_id] }
+    elsif opts[:creation_token] && !opts[:creation_token].empty?
+      raise ArgumentError, "#{@__resource_name__}: creation_token must be minimum 1 and maximum 64-character long." unless opts[:creation_token].length.between?(1, 64)
+      file_system_arguments = { creation_token: opts[:creation_token] }
+    else
+      raise ArgumentError, "#{@__resource_name__}:  `file_system_id` or `creation_token` must be provided."
+    end
+
+    super(opts)
+    validate_parameters(required: [opts.keys[0]])
```

Comment:
Think this is a slightly nicer way of validating the params `validate_parameters(require_any_of: %i(file_system_id creation_token))`
This is a good example: https://github.com/inspec/inspec-aws/blob/master/libraries/aws_iam_policy.rb#L20

Reply:
The `describe_file_systems` method requires only one parameter. Either `file_system_id` or `creation_token`. When both are provided, the validation should fail. The `require_any_of` will pass if provided more than one valid parameters. Refactored code to clarify this: https://github.com/inspec/inspec-aws/pull/198/commits/e7f50d6ac72be65ebb152881feb14c669f193512